doc/iterations/2023.10/shapes/reference_widget_customizability.adoc

```diff
@@ -0,0 +1,79 @@
+= (M) Add customizability to the reference widget
+
+== Problem
+
+As an Application integrator, I would like to control the way the reference widget handles the reference. 
+For instance, I would like to control how the candidate list is built or how the widget's actions are performed.
+
+
+== Key Result
+
+Offer developers the ability to specify in the View DSL all the reference widget customization key points.
+
+== Solution
+
+The approach is to introduce two new advanced reference widgets (one for multi-valued and one for mono-valued) with all customization key points. Once this widget is available it will be easy to convert the simple EMF existing reference widget to use these base widgets.
+
+Here is the model of both mono-valued and multi-valued reference widgets:
+
+image::images/reference_widget_customizability_model.png[Model]
+
+=== Shared property details
+
+The following properties are shared in both widgets.
+
+*_Owner Expression_* (optional): If specified, the reference edited will be searched on the value of the expression, instead of the current element (self)
+
+*_Type Expression_*: The type of the referenced object(s) is obtained by the value of the expression.
+
+*_Candidates Expression_*: This expression is used to express the list of existing compatible objects already present in the editing context (project). These objects can be displayed to allow the user to select one or several of them.
+
+*_Click Operation_* (Optional): If specified, this is the operation that is called when the user clicks a reference value element.
```

Comment:
All of your operations do not have the same shape as existing operations of the view DSL, see:

- ReferenceWidgetDescription#getBody()
- Tool#getBody()
- CheckboxDescription#getBody()